Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaDayFragment.kt

```diff
@@ -93,9 +145,96 @@ class AgendaDayFragment : Fragment(), FlexibleAdapter.OnItemClickListener {
         }
     }
 
+    private fun setupSmoothScroller(): RecyclerView.SmoothScroller {
+        return object : LinearSmoothScroller(context) {
+            override fun getVerticalSnapPreference(): Int {
+                return SNAP_TO_START
+            }
+
+            override fun calculateSpeedPerPixel(displayMetrics: DisplayMetrics): Float {
+                return MILLISECONDS_PER_INCH / displayMetrics.densityDpi
+            }
+        }
+    }
+
+    private fun updateJumpToCurrentButtonVisibility(isCurrentSessionVisible: Boolean) {
+        if (isCurrentSessionVisible) {
+            fadeOutJumpToCurrentButton()
+        } else {
+            if (totalCurrentSessionCount > 0) {
+                fadeInJumpToCurrentButton()
+            }
+        }
+    }
+
+    private fun fadeOutJumpToCurrentButton() {
+        val viewPropAnimator = scrollToCurrentButton
+            .animate()
+            .alpha(JumpToCurrent.ButtonVisibility.minAlpha)
+            .setDuration(JumpToCurrent.ButtonVisibility.duration)
+            .setInterpolator(DecelerateInterpolator())
+        viewPropAnimator.withEndAction { scrollToCurrentButton.visibility = View.GONE }
+        viewPropAnimator.start()
+    }
+
+    private fun fadeInJumpToCurrentButton() {
+        val viewPropAnimator = scrollToCurrentButton
+            .animate().alpha(JumpToCurrent.ButtonVisibility.maxAlpha)
+            .setDuration(JumpToCurrent.ButtonVisibility.duration)
+            .setInterpolator(DecelerateInterpolator())
+        viewPropAnimator.withStartAction { scrollToCurrentButton.visibility = View.VISIBLE }
+    }
+
+    private fun addFloatingAnimation() {
+
+        // Float up
+        val propertyValuesHolder = PropertyValuesHolder.ofFloat(
+            View.TRANSLATION_Y,
+            JumpToCurrent.ButtonTranslation.translationY,
+            -JumpToCurrent.ButtonTranslation.translationY
+        )
+
+        val floatUpAnimator = ObjectAnimator.ofPropertyValuesHolder(
+            scrollToCurrentButton, propertyValuesHolder
+        )
+        floatUpAnimator.duration = JumpToCurrent.ButtonTranslation.duration
+        floatUpAnimator.interpolator = LinearInterpolator()
+
+        // Float down
+        val downFloatValues = PropertyValuesHolder.ofFloat(
+            View.TRANSLATION_Y,
+            -JumpToCurrent.ButtonTranslation.translationY,
+            JumpToCurrent.ButtonTranslation.translationY
+        )
+        val floatDownAnimator = ObjectAnimator.ofPropertyValuesHolder(
+            scrollToCurrentButton, downFloatValues
+        )
+        floatDownAnimator.duration = JumpToCurrent.ButtonTranslation.duration
+        floatDownAnimator.interpolator = LinearInterpolator()
+
+        val floatAnimation = AnimatorSet()
+        floatAnimation.playSequentially(floatUpAnimator, floatDownAnimator)
+        floatAnimation.start()
+        floatAnimation.addListener(object : Animator.AnimatorListener {
```

Comment:
try using AnimatorListenerAdapter ....you won't have to list as many overrides :-)

Reply:
Good call! :-)   I updated only that, retested and pushed.    Once build completes, I will merge.